File: python2-code-samples/putRequest.py
import urllib2, urllib, base64
 
company = "YOUR_TEAMWORK_SITE_NAME"
key = "YOUR_API_KEY"
task_id = TASK_ID
action = "todo_items/{0}/complete.json".format(task_id)
 
request = urllib2.Request("https://{0}.teamwork.com/{1}".format(company, action))
request.add_header("Authorization", "BASIC " +  base64.b64encode(key + ":xxx"))
request.add_header("Content-Type", "application/json")
request.get_method = lambda: "PUT"
 
response = urllib2.urlopen(request)
data = response.read()
print data

File: jQuery/putRequest.js
var company = "YOUR_TEAMWORK_SITE_NAME";
var key = "YOUR_API_KEY";
var task_id = "TASK_ID";
var action = "todo_items/" + task_id + ".json";
var new_name = "This task was edited via a PUT AJAX request sent in JSON format.";

var json = {"todo-item": { "content": new_name } };

$.ajax({
	type: "PUT",
	url: 'https://' + company + '.teamwork.com/' + action,
	headers: {"Authorization": "BASIC " + window.btoa(key + ":xxx")},
	contentType: "application/json; charset=UTF-8",
	data: JSON.stringify(json)
});
